pkg/proxy/iptables/proxier.go

```diff
@@ -280,7 +292,7 @@ func CleanupLeftovers(ipt utiliptables.Interface) (encounteredError bool) {
 		// Hunt for service and endpoint chains.
 		for chain := range existingNATChains {
 			chainString := string(chain)
-			if strings.HasPrefix(chainString, "KUBE-SVC-") || strings.HasPrefix(chainString, "KUBE-SEP-") {
+			if strings.HasPrefix(chainString, "KUBE-SVC-") || strings.HasPrefix(chainString, "KUBE-SEP-") || strings.HasPrefix(chainString, "KUBE-XLB-") {
```

Comment:
I think this PR can move the MARK-MASQ logic from KUBE-FW to KUBE-LB. 

Then KUBE-FW is truly a firewall chain.